pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -550,3 +552,30 @@ func (i *InfraProvider) collectManifests(ctx context.Context, cl client.Client)
 	}
 	return fileList, errorList
 }
+
+func machineIsReady(machine *clusterv1.Machine, publish types.PublishingStrategy, infraID string) (bool, error) {
+	if machine.Status.Phase != string(clusterv1.MachinePhaseProvisioned) &&
+		machine.Status.Phase != string(clusterv1.MachinePhaseRunning) {
+		return false, nil
+	} else if machine.Status.Phase == string(clusterv1.MachinePhaseFailed) {
+		return false, fmt.Errorf("machine %s failed to provision: %q", machine.Name, *machine.Status.FailureMessage)
```

Comment:
I know it was like this before, but do we want to be extra safe and:
```suggestion
		return false, fmt.Errorf("machine %s failed to provision: %q", machine.Name, ptr.Deref(machine.Status.FailureMessage, "no failure message available"))
```

Reply:
pushed